pkg/appgw/requestroutingrules.go

```diff
@@ -38,6 +38,8 @@ func (c *appGwConfigBuilder) getURLPathMaps(cbCtx *ConfigBuilderContext) map[lis
 	for _, ingress := range cbCtx.IngressList {
 		defaultAddressPoolID := c.appGwIdentifier.addressPoolID(defaultBackendAddressPoolName)
 		defaultHTTPSettingsID := c.appGwIdentifier.httpSettingsID(defaultBackendHTTPSettingsName)
+		usePrivateIP, _ := annotations.UsePrivateIP(ingress)
+		usePrivateIP = usePrivateIP || cbCtx.EnvVariables.UsePrivateIP == "true"
```

Comment:
It might be good to drop a log message here that says - "Using Private IP address as requested from {environment / annotation}" -- so we know how `usePrivateIP` was activated.